modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/fsm/QueryExecutor.java

```diff
@@ -620,4 +622,14 @@ static class ParsedResultWithNextCursorFuture {
             this.nextCursorFuture = nextCursorFuture;
         }
     }
+
+    /**
+     * Dumps component state. This method is used for debugging purposes only.
+     */
+    @TestOnly
+    public void dumpComponentState() {
+        if (executionService instanceof ExecutionServiceImpl) {
+            ((ExecutionServiceImpl<?>) executionService).dumpComponentState();
```

Comment:
query may stuck on OPTIMIZATION or CURSOR_INITIALIZATION phases. Such query won't be covered by `ExecutionService`. Let's also dump `runningQueries` collection

Reply:
Fixed